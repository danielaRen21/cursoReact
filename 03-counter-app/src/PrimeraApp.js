import PropTypes from "prop-types";

// FC

//import { json } from "body-parser";

//import React, { Fragment } from "react";

const PrimeraApp = ({
    saludo,
    subtitulo }) => {



    return (
        <>
            <h1>{saludo}</h1>
            {/* <pre>{JSON.stringify(saludo, null, 3)}</pre> */}
            <p>{subtitulo}</p>
        </>

    );


}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}
PrimeraApp.defaultProps = {
    subtitulo: 'hola'
}

export default PrimeraApp;
