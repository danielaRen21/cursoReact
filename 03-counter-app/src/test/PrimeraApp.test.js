import React from 'react';
import PrimeraApp from "../PrimeraApp";
import { shallow } from 'enzyme';
import '@testing-library/jest-dom/extend-expect';
import "../setupTest";

describe('PrimeraApp', () => {
    test('Debe de mostarr el mensaje "Hola soy dani"', () => {
        //const saludo = 'hola soy dani';
        // const { getByText } = render(<PrimeraApp saludo={saludo} />);
        // expect(getByText(saludo)).toBeInTheDocument();
        // //wrapper.getByText()
        const saludo = 'Daniela';
        const wrapper = shallow(<PrimeraApp saludo={saludo} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('should show the subtitle ', () => {
        const saludo = 'Daniela';
        const subtitulo = 'Fernanda'
        const wrapper = shallow(
            <PrimeraApp
                saludo={saludo}
                subtitulo={subtitulo}
            />
        );

        const textoParrafo = wrapper.find('p').text();
        console.log(textoParrafo);
        expect(textoParrafo).toBe(subtitulo);

    })


})
