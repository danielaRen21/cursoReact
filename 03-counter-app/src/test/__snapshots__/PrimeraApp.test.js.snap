// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimeraApp Debe de mostarr el mensaje "Hola soy dani" 1`] = `
<Fragment>
  <h1>
    Daniela
  </h1>
  <p>
    hola
  </p>
</Fragment>
`;
